src/libostree/ostree-repo.c

```diff
@@ -2094,6 +2094,27 @@ append_remotes_d (OstreeRepo          *self,
 }
 
 gboolean
+ostree_repo_is_created (OstreeRepo    *self,
+                        gboolean      *does_exist,
+                        GError        **error)
+{
+  struct stat stbuf;
+
+  if (lstat (gs_file_get_path_cached (self->objects_dir), &stbuf) < 0)
```

Comment:
So just to clarify, you meant to keep ostree_repo_exists() as is and add another function called ostree_repo_is_created() that checks if the repo exists and then checks to make sure it can be opened?